Add unmarshal and raw unmarshal

garrison-henkle opened the pull request:
This PR is a fork of @TimothyStiles's #3 that adds an UnmarshalRaw function that can be used on raw SurrealQL responses returned by the Query function. A couple tests were also added for the new function.

I'm fairly new to Go, so I'm not too familiar with the style of Go code. If my changes do not seem to follow standard Go style, I would be happy to refactor the PR.

TimothyStiles commented on db_test.go:
```diff
@@ -1,75 +1,230 @@
+
+	// unmarshal the data into a user slice
+	var users []testUser
+	err = surrealdb.Unmarshal(userData, &users)
```
should this be `surrealdb.UnmarshalRaw(userData, &users)`?

TimothyStiles reviewed (approved):
@garrison-henkle forked pr #3 and improved its functionality. Instead of merging #3 can we try to review and merge this instead @tobiemh?

garrison-henkle commented:
I merged in recent changes from main and cleaned up a few things that shouldn't have been in there (my personal .gitignore additions, a useless call I added into one of the tests, etc).

I also changed all the `any`s to `interface{}`s to match the changes from #9. I think the errors being caused by `any` in #9 are caused by an old compiler (`any` was introduced in 1.18), but `interface{}` is probably better anyway for compatibility.

The switch / select statement indentation changes from #9 may not be the recommended Go style considering my formatter keeps trying to change them, but I kept them in there for the sake of not changing main.

tobiemh commented:
Hey @garrison-henkle if you could rebase from main again. Then I'll get this merged!

garrison-henkle commented:
@tobiemh It should be good now!

TimothyStiles commented:
@tobiemh +1 for merge! I'm excited for this one.
